invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/UppyUploader/UppyUploader.js

```diff
@@ -125,31 +153,17 @@ export const UppyUploaderComponent = ({
   );
 
   React.useEffect(() => {
-    const onFileAdded = (file) => {
-      const normalizedName = file.name?.normalize?.() ?? file.name;
-      const alreadyUploaded =
-        normalizedName &&
-        filesList.some(
-          (item) =>
-            item.name?.normalize?.() === normalizedName && !item.uploadState.isFailed
-        );
-
-      if (alreadyUploaded) {
-        uppy.info(
-          i18next.t("{{filename}} is already part of this upload.", {
-            filename: file.name,
-          }),
-          "warning"
-        );
-        uppy.removeFile(file.id);
-      }
-    };
+    filesListRef.current = filesList;
+  }, [filesList]);
 
+  React.useEffect(() => {
+    const onFileAdded = (file) =>
+      warnOnDuplicateFileAdded(uppy, file, filesListRef.current);
     uppy.on("file-added", onFileAdded);
```

Comment:
https://uppy.io/docs/uppy/#onbeforefileaddedfile-files Uppy option might be more suittable for this case - it seems you can run the check and just return false for duplicates, without a need for removing already added files to uppy state